scrapy/crawler.py

```diff
@@ -221,6 +226,9 @@ def join(self):
         while self._active:
             yield defer.DeferredList(self._active)
 
+    def _is_spider_created_for_every_crawler(self):
+        return all(self.crawlers_has_spiders)
```

Comment:
I'd like to propose the removal of all this methods and use a single boolean flag that summarizes the outcome of bootstraping the spider. 

The public attribute name would be `bootstrap_failed` and we set it at `_crawl / _done()` like:
```python
    self.bootstrap_failed |= not getattr(crawler, 'spider', None)
```
this public attribute replaces `_is_spider_created_for_every_crawler()` method and `crawlers_has_spiders` atttribute, but also remove the need for private `_is_spider_created()` method.   

then the check in `runspider.py` and `crawl.py` looks like:

```python
if not self.crawler_process.bootstrap_failed:
    self.exitcode = 1
```